### ISSUE TITLE
Composer prevents loss from converging when using DecoupledAdamW

### ISSUE THREAD

**Issue by vedantroy** [mosaicml/composer#1334]
** Environment **
- OS: Ubuntu 22.04 LTS
- GPU:
```
  *-display                 
       description: VGA compatible controller
       product: GP102 [GeForce GTX 1080 Ti]
       vendor: NVIDIA Corporation
       physical id: 0
       bus info: pci@0000:01:00.0
       version: a1
       width: 64 bits
       clock: 33MHz
       capabilities: pm msi pciexpress vga_controller bus_master cap_list rom
       configuration: driver=nvidia latency=0
       resources: irq:58 memory:fa000000-faffffff memory:c0000000-cfffffff memory:d0000000-d1ffffff ioport:e000(size=128) memory:c0000-dffff
  *-graphics
       product: EFI VGA
       physical id: 2
       logical name: /dev/fb0
       capabilities: fb
       configuration: depth=32 resolution=1024,768
```
- Cuda: 11.7
- Composer: 0.8.2

I'm trying to implement a paper in Pytorch, using composer for my training loop.
See here: https://github.com/vedantroy/improved-ddpm-pytorch/commit/24909c064953bf54bf3a50c64923fdfef714f3cc

Composer is breaking stuff in at least 2 different ways:
1. If I replace this line with composer's `DecoupledAdamW` then the loss goes from 1.0 => 0.6 but not much further down
   https://github.com/vedantroy/improved-ddpm-pytorch/blob/24909c064953bf54bf3a50c64923fdfef714f3cc/openai_iddpm.py#L103
2. ~If I use the `OpenAIIDDPM` model instead of the `manual_train` function, the loss does not go down from around 1.0 at all
  https://github.com/vedantroy/improved-ddpm-pytorch/blob/24909c064953bf54bf3a50c64923fdfef714f3cc/openai_iddpm.py#L72~
 - I think only the 1st thing was a bug b/c once I switched to torch's `AdamW` and removed the warmup, the learning rate progressed as normal

I've uploaded the data here, so you can reproduce it: https://drive.google.com/drive/folders/1zK-3SkGOETGFajIBSXbEPqPSRdwqajUq?usp=sharing

It's just the CelebHQ dataset in 64x64 resolution, stored as parquet. Any image dataset should work and you can probably swap out my torch data based dataloader for something more standard.

**Comment by abhi-mosaic** [comment]
Hey, thank you for raising this issue!

I think this is a bug on our side where we copied over torch's `AdamW` defaults, in particular the `weight_decay=1e-2`.

Since our decoupled optimizers [implement the original paper](https://arxiv.org/abs/1711.05101) and completely separate `lr` and `weight_decay`, the default of `1e-2` is much too high, and would decay your weights by 1/100th every batch! This is likely what was causing the poor convergence. I think our internal benchmarks that used `DecoupledAdamW` always set `weight_decay` explicitly which is why we missed this bug :( 

I will update the default value to `weight_decay=1e-5`, which should be close if not identical to the behavior of torch's `AdamW` which has defaults `lr=1e-3` and `weight_decay=1e-2`.
